src/MagnumPlugins/UfbxImporter/UfbxImporter.cpp

```diff
@@ -0,0 +1,1287 @@
+	case UFBX_BLEND_SCREEN: return "screen"_s;
+	case UFBX_BLEND_COLOR_DODGE: return "colorDodge"_s;
+	case UFBX_BLEND_LINEAR_DODGE: return "linearDodge"_s;
+	case UFBX_BLEND_LIGHTER_COLOR: return "lighterColor"_s;
+	case UFBX_BLEND_SOFT_LIGHT: return "softLight"_s;
+	case UFBX_BLEND_HARD_LIGHT: return "hardLight"_s;
+	case UFBX_BLEND_VIVID_LIGHT: return "vividLight"_s;
+	case UFBX_BLEND_LINEAR_LIGHT: return "linearLight"_s;
+	case UFBX_BLEND_PIN_LIGHT: return "pinLight"_s;
+	case UFBX_BLEND_HARD_MIX: return "hardMix"_s;
+	case UFBX_BLEND_DIFFERENCE: return "difference"_s;
+	case UFBX_BLEND_EXCLUSION: return "exclusion"_s;
+	case UFBX_BLEND_SUBTRACT: return "subtract"_s;
+	case UFBX_BLEND_DIVIDE: return "divide"_s;
+	case UFBX_BLEND_HUE: return "hue"_s;
+	case UFBX_BLEND_SATURATION: return "saturation"_s;
+	case UFBX_BLEND_COLOR: return "color"_s;
+	case UFBX_BLEND_LUMINOSITY: return "luminosity"_s;
+	case UFBX_BLEND_OVERLAY: return "overlay"_s;
+    default:
+        Warning{} << "Unhandled blend mode" << Int(mode);
+        return {};
+    /* LCOV_EXCL_STOP */
+    }
+}
+
+ufbx_load_opts loadOptsFromConfiguration(Utility::ConfigurationGroup& conf) {
+    ufbx_load_opts opts = {};
+
+    const Long maxTemporaryMemory = conf.value<Long>("maxTemporaryMemory");
+    const Long maxResultMemory = conf.value<Long>("maxResultMemory");
+    const bool geometricTransformNodes = conf.value<bool>("geometricTransformNodes");
+    const bool preserveRootNode = conf.value<bool>("preserveRootNode");
+    const bool normalizeUnits = conf.value<bool>("normalizeUnits");
+
+    opts.generate_missing_normals = conf.value<bool>("generateMissingNormals");
+    opts.strict = conf.value<bool>("strict");
+    opts.disable_quirks = conf.value<bool>("disableQuirks");
+    opts.load_external_files = conf.value<bool>("loadExternalFiles");
+    opts.ignore_geometry = conf.value<bool>("ignoreGeometry");
+    opts.ignore_animation = conf.value<bool>("ignoreAnimation");
+    opts.ignore_embedded = conf.value<bool>("ignoreEmbedded");
+    opts.ignore_all_content = conf.value<bool>("ignoreAllContent");
+    opts.ignore_missing_external_files = true;
+
+    /* Substitute zero maximum memory to one, so that if the user computes the
+       maximum memory and ends up with zero it doesn't result in unlimited */
+    if(maxTemporaryMemory >= 0)
+        opts.temp_allocator.memory_limit = size_t(Utility::max(maxTemporaryMemory, Long(1)));
+    if(maxResultMemory >= 0)
+        opts.result_allocator.memory_limit = size_t(Utility::max(maxResultMemory, Long(1)));
+
+    /* By default FBX has cameras pointing at +X and lights at -Y, let ufbx
+       normalize those to what Magnum expects */
+    opts.target_light_axes = ufbx_axes_right_handed_y_up;
+    opts.target_camera_axes = ufbx_axes_right_handed_y_up;
+
+    /* @todo expose more of these as options? need to think of reasonable
+       defaults anyways, feels like ignoring geometry transform is not great */
+    if(preserveRootNode)
```

Comment:
This collates `preserveRootNode` with writing the potential unit/coordinate conversion to the root node. Not sure if this is smart here but I don't really see any other use for `preserveRootNode` anymore, unless we just want to get rid of that configuration.

Reply:
I'd vote for removing that option and making it implicit (my [other comment for reference](https://github.com/mosra/magnum-plugins/pull/133#discussion_r1045270558)).

Reply:
Sounds like a good idea, there's no real use for it unless you're writing the coordinate/unit transform to it.